components/node-labeler/cmd/run.go

```diff
@@ -249,6 +282,184 @@ func (r *PodReconciler) Reconcile(ctx context.Context, req reconcile.Request) (r
 	return reconcile.Result{}, nil
 }
 
+type WorkspaceCountController struct {
+	client.Client
+	nodesToReconcile chan string
+	stopChan         chan struct{}
+}
+
+func NewWorkspaceCountController(client client.Client) (*WorkspaceCountController, error) {
+	return &WorkspaceCountController{
+		Client:           client,
+		nodesToReconcile: make(chan string, 1000),
+		stopChan:         make(chan struct{}),
+	}, nil
+}
+
+func (wc *WorkspaceCountController) SetupWithManager(mgr ctrl.Manager) error {
+	// Start the periodic reconciliation goroutine
+	go wc.periodicReconciliation()
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("workspace-count").
+		For(&workspacev1.Workspace{}).
+		WithEventFilter(wc.workspaceFilter()).
+		Complete(wc)
+}
+
+func (wc *WorkspaceCountController) periodicReconciliation() {
```

Comment:
We do a manual reconciliation every 5m. Why? Because `SyncPeriod` in the manager options doesn't seem to really be doing what I expected it to do - I didn't see the reconciliation trigger via an external trigger even once during testing.

Hence the goroutine, which we properly dispose of with the `stopChan`